consensus/src/sync/pico/sync_stream.rs

```diff
@@ -133,28 +133,33 @@ impl<TNetwork: Network> PicoMacroSync<TNetwork> {
                                     BlockchainProxy::Light(ref light_blockchain) => {
                                         let blockchain = light_blockchain.upgradable_read();
                                         let latest_block_number = blockchain.block_number();
+                                        let latest_election_head =
+                                            blockchain.election_head.block_number();
 
                                         if block.block_number() < latest_block_number {
-                                            // The peer is outdated, so we emit it, and we remove it
-                                            self.peer_requests.remove(&peer_id);
-                                            self.syncing_peers.remove(&peer_id);
-                                            return Poll::Ready(Some(MacroSyncReturn::Outdated(
-                                                peer_id,
-                                            )));
+                                            // We could be in an more recent state at this point so we have to process the next block.
```

Comment:
I guess this is in the context of us getting an older election block because we have reached the maximum number of epochs that can be returned by the macro chain request?